### PR TITLE
On export, mark all variables as having mode In_types in Aliases

### PR DESCRIPTION
**Pull request by lthls**
Split from #299

### CODE REVIEW

File: middle_end/flambda/types/env/aliases.ml [hunk 1]
```diff
@@ -595,3 +621,29 @@ let get_canonical_ignoring_name_mode t name =
...
+        let new_mode =
+          if Simple.is_var simple then Name_mode.in_types
+          else BTM.name_mode binding_time_and_mode
+        in
+        BTM.create (BTM.binding_time binding_time_and_mode) new_mode)
+      binding_times_and_modes
+  in
+  let aliases_of_canonical_elements =
+    (* Note: the relative order of the aliases and of their canonical element
+       will be unchanged, as it only compares the binding times.
```
Comment (mshinwell) [inline middle_end/flambda/types/env/aliases.ml]: Please clarify what "it" refers to.

File: middle_end/flambda/types/env/typing_env.rec.ml [hunk 2]
```diff
@@ -436,16 +438,19 @@ let print_with_cache ~cache ppf
       Scope.Map.filter (fun _ level -> not (One_level.is_empty level))
         levels
     in
+    let aliases = Cached.aliases (One_level.just_after_level current_level) in
```
Comment (mshinwell) [inline middle_end/flambda/types/env/typing_env.rec.ml]: These changes to the printers should be in a separate PR please and guarded by a command-line flag.

**Comment by mshinwell** [comment]
OK to merge after two changes.

File: middle_end/flambda/types/env/typing_env.rec.ml (hunk 2 above)
Comment (lthls) [reply middle_end/flambda/types/env/typing_env.rec.ml]: I don't agree with the command-line flag part, particularly when so much information is printed already.
But I'll push this change into a separate PR, so that it doesn't block this one.

File: middle_end/flambda/types/env/aliases.ml (hunk 1 above)
Comment (lthls) [reply middle_end/flambda/types/env/aliases.ml]: The "it" was supposed to refer to the order, but I agree that it's not obvious. Would "as it only depends on the binding times" be clearer ?

Comment (mshinwell) [reply middle_end/flambda/types/env/aliases.ml]: Yes, fine.

Comment (lthls) [reply middle_end/flambda/types/env/aliases.ml]: I've updated the branch with this version (and without the printing changes) in the meantime, so you should be able to merge directly.
